alacritty_terminal/src/term/mod.rs

```diff
@@ -1855,12 +1862,17 @@ impl<T: EventListener> Handler for Term<T> {
             let graphic_cell = GraphicCell { texture: texture.clone(), offset_x: 0, offset_y };
             self.grid[line][Column(left)].set_graphic(graphic_cell);
 
-            if scrolling {
+            if scrolling && offset_y < height - self.cell_height as u16 {
                 self.linefeed();
             }
         }
 
-        if scrolling {
+        if self.mode.contains(TermMode::SIXEL_CURSOR_TO_THE_RIGHT) {
+            let graphic_columns = (graphic.width + self.cell_width - 1) / self.cell_width;
+            let right = min(self.columns(), left + graphic_columns);
+            self.move_forward(Column(right));
```

Comment:
`move_forward` appends its parameter to the current position of the cursor, and `right` in this code already contains that position (the `left` variable). I have not tested it, but it seems that, if the graphic is inserted in a position different to column 0, the final position will be wrong.

Something like `self.move_forward(Column(graphic_columns))` should be enough.

Reply:
Good catch; I had misunderstood that. I’ve fixed it:

![Screenshot from 2021-06-20 20-20-39](https://user-images.githubusercontent.com/228849/122702808-b700fd00-d23f-11eb-979e-c1ed871ec80b.png)